test/unit/account/search_test.rb

```diff
@@ -77,6 +77,10 @@ class Account::SearchTest < ActiveSupport::TestCase
         buyers = provider.buyer_accounts.scope_search(:query => 'foo@bar.co.example.com')
         expected_buyer = User.find_by!(email: 'foo@bar.co.example.com').account
         assert_equal [expected_buyer], buyers
+
+        # incremental index update check
+        provider.smart_destroy
+        assert_equal [], provider.buyer_accounts.scope_search(:query => 'foo@bar.co.example.com')
```

Comment:
This is totally expected because the buyer has been removed. What we need to assert is whether the index still exists.
I've been using this to get the number of indexed records:
```
conn = ThinkingSphinx::Connection.new
res = conn.execute("show index account_core status")
res.map(&:values).to_h["indexed_documents"]
```
Maybe there's a way to query the index directly but at least we should assert that `indexed_documents` is short by the number of buyers deleted (2 in this case).

Reply:
This is searching the index. If my indexer changes are not applied, then the record is still found.

Reply:
Anyway, updated to catch regardless of record existence.